lib/ansible/modules/windows/win_pagefile.ps1

```diff
@@ -0,0 +1,153 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+########
+
+Function Remove-Pagefile($path, $whatif) 
+{
+    Get-WmiObject Win32_PageFileSetting | WHERE { $_.Name -eq $path } | Remove-WmiObject -WhatIf:$whatif
+}
+
+Function Get-Pagefile($path)
+{
+    Get-WmiObject Win32_PageFileSetting | WHERE { $_.Name -eq $path }
+}
+
+########
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name '_ansible_check_mode' -type 'bool' -default $false
+
+$automatic = Get-AnsibleParam -obj $params -name "automatic" -type "bool"
+$drive = Get-AnsibleParam -obj $params -name "drive" -type "str"
+$fullPath = $drive + ":\pagefile.sys"
+$initialSize = Get-AnsibleParam -obj $params -name "initial_size" -type "int"
+$maximumSize = Get-AnsibleParam -obj $params -name "maximum_size" -type "int"
+$override =  Get-AnsibleParam -obj $params -name "override" -type "bool" -default $true
+$removeAll = Get-AnsibleParam -obj $params -name "remove_all" -type "bool" -default $false
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "query" -validateset "present","absent","query"
+$systemManaged = Get-AnsibleParam -obj $params -name "system_managed" -type "bool" -default $false
+$testPath = Get-AnsibleParam -obj $params -name "test_path" -type "bool" -default $true
+
+$result = @{
+    changed = $false
+}
+
+if ($removeAll) {
+    $currentPageFiles = Get-WmiObject Win32_PageFileSetting
+    if ($currentPageFiles -ne $null) {
+        $currentPageFiles | Remove-WmiObject -WhatIf:$check_mode | Out-Null
+        $result.changed = $true
+    }
+}
+
+if ($automatic -ne $null) {
+    # change autmoatic managed pagefile 
+    try {
+        $computerSystem = Get-WmiObject -Class win32_computersystem -EnableAllPrivileges
+    
+        if ($computerSystem.AutomaticManagedPagefile -ne $automatic) {
+            $computerSystem.AutomaticManagedPagefile = $automatic
+            if (-not $check_mode) {
+            	$computerSystem.Put() | Out-Null
+            }
+            $result.changed = $true
+        }
+    } catch {
+        Fail-Json $result "Failed to set AutomaticManagedPagefile $_.Exception.Message"
+    }
+}
+
+if ($state -eq "absent") {
+    # Remove pagefile
+    if ((Get-Pagefile $fullPath) -ne $null)
+    {
+        try {
+            Remove-Pagefile $fullPath -whatif:$check_mode
+            $result.changed = $true
+        } catch {
+            Fail-Json $result "Failed to remove pagefile $_.Exception.Message"
+        }
+    }
+} elseif ($state -eq "present") {   
+    # Remove current pagefile
+    if ($override) {
+        if ((Get-Pagefile $fullPath) -ne $null)
+        {
+            try {
+                Remove-Pagefile $fullPath -whatif:$check_mode
+                $result.changed = $true
+            } catch {
+                Fail-Json $result "Failed to remove current pagefile $_.Exception.Message"
```

Comment:
You need to enclose the exception part in brackets like `$($_.Exception.Message)`